### PR TITLE
feat: containerized entire app, and built simple front-end

### PR DESCRIPTION
**Pull request by deltabrot**
I have also removed a lot of unused code.

To run the entire stack with hot reloading, you can now just run:
```bash
./run.sh
```

While in the root of the repo.

### CODE REVIEW

File: api/app.py [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+# init
 dotenv.load_dotenv()
 app = Flask(__name__)
+cors = CORS(app, resources={r"/*": {"origins": "http://localhost:3000"}})
 
+# create database
 create_database()
 
-# VALIDATED
+print(f'Listening on http://0.0.0.0:5000', flush=True)
```
Comment (djermy) [inline api/app.py:26]: What purpose does this serve? presumably it's for logging when you run the app?

File: api/database/database.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from api.service.runescape.json_cleaner import filter_item_details
+import sqlite3
+
+JSON_PATH = 'service/runescape/items_json/'
+DB_PATH = 'database/'
+
+
```
Comment (djermy) [inline api/database/database.py]: white space, is this file to replace the original database_handler and it's create_database function?

File: api/requirements.txt [hunk 3]
```diff
@@ -7,27 +7,21 @@ charset-normalizer==2.0.4
...
 itsdangerous==2.0.1
 Jinja2==3.1.2
 MarkupSafe==2.1.1
-mkl-fft==1.3.6
-mkl-random==1.2.2
-mkl-service==2.4.0
 numexpr==2.8.4
-numpy==1.24.3
-pandas==1.5.3
-pip==23.0.1
```
Comment (djermy) [inline api/requirements.txt:20]: Have you confirmed all end points and associated calculators work without these packages?
Also why did you remove pip, will that not cause issues when trying to install new packages?

File: front-end/public/logo192.png
Comment (djermy) [inline front-end/public/logo192.png:1]: What are these images for?

**Review by djermy — COMMENTED** [review]
Thank you for the contribution. I've had a look at the files you changed and deleted. there are just a few things I'd like to clarify

File: api/app.py (hunk 1 above)
Comment (deltabrot) [reply api/app.py:26]: It just prints the API url each time the add reloads. When the app hot reloads, you lose the original output showing this so I thought I'd reprint.

File: api/database/database.py (hunk 2 above)
Comment (deltabrot) [reply api/database/database.py]: Feel free to remove the white space. This is `database_handler`, just renamed and with all of the unused methods removed.

File: api/requirements.txt (hunk 3 above)
Comment (deltabrot) [reply api/requirements.txt:20]: No I didn't confirm this, feel free to check it out yourself.

Probably the best thing would be to write some unit tests for your endpoints to make sure they're all still functional whenever code changes are made.

File: front-end/public/logo192.png
Comment (deltabrot) [reply front-end/public/logo192.png:1]: They're just the default react app images. Feel free to remove.

Comment (djermy) [reply front-end/public/logo192.png:1]: is there any reason to have them? do they show anywhere to the end user?
